File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int result = 0;

        for (int n : nums) {
            if (!set.contains(n)) {
                set.add(n);
            }
        }
        
        for (int n : nums) {
            if (set.contains(n - 1)) {
                continue;
            }
            int count = 1;
            int i = n;
            while (set.contains(i++)) {
                result = Math.max(result, count++);
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if not nums:
            return 0
        if len(nums) == 1:
            return nums[0]
            
        nums1 = nums[:-1]
        nums2 = nums[1:]

        def dp(nums, mp = {}):
            print(nums)
            mp[0] = nums[0]

            for i in range(1, len(nums)):
                mp[i] = max(nums[i] + mp.get(i - 2, 0), mp[i - 1])

            return mp[len(mp) - 1]

        return max(dp(nums1), dp(nums2))
                
